doc/Language/classtut.pod6

```diff
@@ -5,86 +5,60 @@
 =SUBTITLE A tutorial about creating and using classes in Raku
 
 X<|OOP>
+Raku provides a rich built-in syntax for defining and using classes. It makes writing classes
+expressive and short for most cases, but also provides mechanisms to cover
+the rare corner cases.
 
-Raku has a rich built-in syntax for defining and using classes.
+=head1 A quick overview
 
-A default constructor allows the setting of attributes for the created
-object:
+Let's start with an example to give an overview:
 
 =begin code
-class Point {
-    has Int $.x;
-    has Int $.y;
-}
-
 class Rectangle {
-    has Point $.lower;
-    has Point $.upper;
+    has Int $.a = 1;
+    has Int $.b = 1;
+
+    submethod TWEAK(:$log) {
+        say "Constructing Rectangle ($!a|$!b)" if $log;
+    }
 
     method area(--> Int) {
-        ($!upper.x - $!lower.x) * ( $!upper.y - $!lower.y);
+        return $!a * $!b;
     }
 }
 
-# Create a new Rectangle from two Points
-my $r = Rectangle.new(lower => Point.new(x => 0, y => 0),
-                      upper => Point.new(x => 10, y => 10));
+my $r1 = Rectangle.new(a => 2, b => 3);
+say $r1.area(); # OUTPUT: «6␤»
 
-say $r.area(); # OUTPUT: «100␤»
+my $r2 = Rectangle.new(b => 2, :log); # OUTPUT: «Constructing Rectangle (1|2)␤»
 =end code
 
-In the two classes, the default constructor is being used. This
-constructor will use named parameters in its invocation: C«Point.new(x =>
-0, y => 0)».
+We define a new I<Rectangle> class using the L<class|/language/objects#Classes> keyword. It has two I<attributes>, C<$!a> and C<$!b> introduced with the C<has> keyword. Both default to C<1>. Read only accessor methods are automatically generated. (Note the C<.> instead of C<!> in the declaration, which triggers the generation. Mnemonic: C<!> resembles a closed door, C<.> an open one.) 
```

Comment:
This is good, and an improvement, but still TWEAK is not mentioned and thus probably left out for the time being.

Reply:
Two additional notes:
1. Please check line length
2. You could mention $.a and $.b make them public, so there's also a mutator thrown in for free.

Reply:
Ah, I didn't notice. `$.a` is introduced, not `$!a`

Reply:
I'll correct line length after all other issues are solved. Otherwise I'll do it over and over again.